lib/mongo/cluster/reapers/cursor_reaper.rb

```diff
@@ -197,7 +197,10 @@ def kill_cursors
           op.execute(server, context: Operation::Context.new(options: options))
 
           if session = kill_spec.session
-            if session.implicit? && !session.ended?
+            if session.ended?
+              raise Error::LintError, "Ended session for a garbage collected cursor"
```

Comment:
This actually needs to be conditioned on linting being enabled (`if Lint.enabled?`)